02_Spatial_data_and_R_packages_for_mapping.Rmd

```diff
@@ -0,0 +1,356 @@
+# Spatial data and R packages for mapping
+
+**Learning objectives:**
+
+- Spatial data types
+- CRS
+- making a map with R
+
+## Spatial data and R packages for mapping
+
+
+Type of Spatial data:
+
+- areal
+- geostatistical
+- point patterns
+
+```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE}
+library(tidyverse)
+```
+
+Packages used:
+
+- ggplot2
+- leaflet
+- mapview
+- tmap
+
+### Types of spatial data
+
+
+Define the space:
+
+$$\left\{Z(s):s\in D \subset \mathbf{R^d} \right\}$$
+
+$$Z=\text{observed values}$$
+$$s=\text{location}$$
+$$D=\text{domain (aereal, geostatistical, point patterns)}$$
+
+$$d=\text{dimension}$$
+
+
+#### Areal data
+
+The domain $D$ is **fixed** (of regular or irregular shape) and made of finite number of areal units with well-defined boundaries:
+
+- zip code
+- census tract
+- pixels
+
+```{r echo=TRUE, message=FALSE, warning=FALSE, paged.print=FALSE}
+require(sf)
+require(ggplot2)
+require(viridis)
+
+nc <- st_read(system.file("shape/nc.shp", package = "sf"),
+  quiet = TRUE
+)
+
+
+
+ggplot(data = nc, aes(fill = SID74)) + 
+  geom_sf() +
+  scale_fill_viridis() + 
+  labs(title="Number of sudden infant deaths",
+       subtitle="counties of North Carolina, USA, in 1974") +
+  theme_bw()
+```
+
+
+#### Geostatistical data
+
+The domain $D$ is a **continuous fixed**:
+
+- air pollution
+- rainfall values
+
+Data represent the average rainfall measurements for the period May-June (dry season) over different years collected at 143 recording stations throughout Paraná state, Brazil.
+
+```{r eval=TRUE, message=FALSE, warning=FALSE, paged.print=FALSE}
+require(geoR)
+
+data.frame(cbind(parana$coords, Rainfall = parana$data))
+
+
+ggplot(data.frame(cbind(parana$coords, Rainfall = parana$data)))+
+  geom_point(aes(east, north, color = Rainfall), size = 2) +
+  coord_fixed(ratio = 1) +
+  scale_color_gradient(low = "blue", high = "orange") +
+  geom_path(data = data.frame(parana$border), aes(east, north)) +
+  labs(title="Average rainfall measurements",
+       subtitle="May-June Paraná state, Brazil")+
+  theme_bw()
+```
+
+
+#### Point patterns
+
+The domain $D$ in point patterns is **random**.
+
+
+```{r}
+library(cholera)
+rng <- cholera::mapRange()
+
+
+plot(fatalities[, c("x", "y")],
+  pch = 15, col = "black",
+  cex = 0.5, xlim = rng$x, ylim = rng$y, 
+  asp = 1,main="The locations of deaths\nof the 1854 London cholera outbreak",
+  frame.plot = FALSE, 
+  axes = FALSE, xlab = "", ylab = ""
+)
+addRoads()
+?cholera::addRoads()
```

Comment:
I'll be interested to see how this behaves in the Rmd, I'm not sure (I'll check after this first round of fixes).